app/src/test/kotlin/Game2048Tests.kt

```diff
@@ -0,0 +1,77 @@
+package Twenty48
+
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.Assertions.assertTrue
+import kotlin.random.Random
+import StatelessSolver
+
+class Game2048Tests {
+
+    val scenarioGrid1 = arrayOf(
+            arrayOf(0, 0, 2, 0),
+            arrayOf(2, 2, 0, 0),
+            arrayOf(64, 64, 2, 2),
+            arrayOf(128, 256, 512, 1024)
+    )
+
+    val scenarioGrid2 = arrayOf(
+            arrayOf(1024, 0, 0, 0),
+            arrayOf(512, 512, 0, 0),
+            arrayOf(0, 0, 0, 0),
+            arrayOf(0, 0, 0, 0)
+    )
+
+    fun testGame2048(inputGrid: Array<Array<Int>>  ): Int? {
+
+        var testGrid = inputGrid.copyOf()
+
+        val initialGameState = Game2048State(Game2048Position(testGrid))
+        val game2048MDP = Game2048MDP(initialGameState)
+
+        var solver = StatelessSolver(
+                game2048MDP,
+                Random,
+                500,
+                10,
+                1.4,
+                0.9,
+                true
+        )
+        solver.buildTree()
+        solver.displayTree()
+
+        println("optimalAction")
+        println(solver.getNextOptimalAction().toString())
+
+        println("optimal Horizon")
+        val solList = solver.getOptimalHorizon().map { it.toString() }
+
+        println(solList)
+        // simply replay the 2048 game using the solution
+
+        var gc = Game2048Controller()
+        for (a in solList){
+            testGrid = gc.manipulateGrid(testGrid, a)
+        }
+        println(Game2048State(Game2048Position(testGrid)).toString())
+        println(Game2048State(Game2048Position(testGrid)).score.toString())
+
+        return Game2048State(Game2048Position(testGrid)).score
+
+    }
+
+    @Test fun game2048ests() {
```

Comment:
Again, one scenario per test. Running multiple tests in one function is usually an anti-pattern.